### PR TITLE
Specify OMR_CUDA_HOME when CUDA is enabled

### PR DESCRIPTION
**Pull request by rwy0717**
The search mechanism for CUDA in OMR is changing. Defining OMR_CUDA_HOME will allow OMR to continue to locate CUDA with the new mechanism.

Signed-off-by: Robert Young <rwy0717@gmail.com>

### CODE REVIEW

File: closed/OpenJ9.gmk [hunk 1]
```diff
@@ -493,7 +493,7 @@ else
 endif
 
 ifeq (true,$(OPENJ9_ENABLE_CUDA))
-  CMAKE_ARGS += -DJ9VM_OPT_CUDA=ON
+  CMAKE_ARGS += -DJ9VM_OPT_CUDA=ON -DOMR_CUDA_HOME="$(CUDA_HOME)"
   CMAKE_CUDA_ENV := CUDA_BIN_PATH="$(CUDA_HOME)"
```
Comment (keithc-ca) [inline closed/OpenJ9.gmk]: We don't need `CMAKE_CUDA_ENV` any more: can you remove its definitions and uses?

Comment (rwy0717) [reply closed/OpenJ9.gmk]: I was thinking, In order to keep the repository working, that could come in a later PR. So we'd accept this PR and it's counterpart in jdk11, then accept the OMR patch, then I'd open two new PRs cleaning up OpenJ9.gmk. That way, there's no window where cmake builds are broken. Does that sound OK?

**Comment by rwy0717** [comment]
The copyright and line endings checks are broken, they both failed to check out the PR. the copyright date is already up-to-date, and this PR is coming from an osx machine, so I doubt there are line ending issues.

Maybe someone can rerun the tests, or we can just ignore the failures?

**Comment by keithc-ca** [comment]
Jenkins copyright check

**Comment by keithc-ca** [comment]
Jenkins line endings check

**Comment by rwy0717** [comment]
Same failure, weird. The tool is trying to check out the wrong commit. Here's some choice output:

The checkout (wrong):
```
git checkout -f 54dee59d2d40b97e2bd605178f7282ae3057aeac
```

Setting the status (correct):
```
Setting status of 64eb96991a68d0cede0f65489932a7f8d81a51df to FAILURE
```

**Comment by keithc-ca** [comment]
Those are the expected commit SHAs (`64eb96991a68d0cede0f65489932a7f8d81a51df` is the head of your branch, `54dee59d2d40b97e2bd605178f7282ae3057aeac` is `refs/pull/merge/374`).

**Comment by keithc-ca** [comment]
Jenkins copyright check

**Comment by keithc-ca** [comment]
Jenkins copyright check

**Comment by keithc-ca** [comment]
Jenkins line endings check

**Comment by keithc-ca** [comment]
Jenkins compile xlinuxcm jdk8

**Pull request merged**